w3-filecoin.md

```diff
@@ -0,0 +1,187 @@
+# Filecoin Protocol
+
+![status:wip](https://img.shields.io/badge/status-wip-orange.svg?style=flat-square)
+
+## Editors
+
+- [Vasco Santos], [Protocol Labs]
+
+## Authors
+
+- [Vasco Santos], [Protocol Labs]
+
+# Abstract
+
+This spec describes the filecoin protocol used by web3.storage's "w3up" platform to interact with Spade ♠️ in order to get user/application CAR files into Filecoin deals.
+
+## Language
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+# Terminology
+
+## Roles
+
+There are several roles in the authorization flow:
+
+| Name        | Description |
+| ----------- | ----------- |
+| Storage Provider | [Principal] identified by [`did:web`] identifier, representing a storage provider like w3up |
+| Authority   | [Principal] that represents service provider that executes invoked capabilities |
+| Verifier   | Component of the [authority] that performs UCAN validation |
+
+### Storage Provider
+
+A _Storage Provider_ is a type of [principal] identified by a [`did:web`] identifier.
+
+A Storage Providers facilitates data storage services to applications and users, getting the requested data stored into Filecoin deals asynchronously.
+
+### Authority
+
+_Authority_ is a [principal] that executes invoked capabilities.
+
+### Verifier
+
+Component of the [authority] that performs UCAN validation
+
+# Protocol
+
+## Overview
+
+A Storage Provider is the entry point for user/application data into the web3. It will act on behalf of users to move data around into different storage points. One of the key storage points is Filecoin deals.
+
+## Authorization
+
+Storage Provider (web3.storage) must be given capabilities by a Filecoin broker Authority (Spade) to invoke capabilities to get its stored content into Filecoin deals.
```

Comment:
I think this makes sense in a "push" world, but in a "pull" world, I think it can be somewhat simplified? In the push world, you need some sort of permission to invoke a method to get what you want. 

But in a pull world, you don't need anyone's permission to make data available via an endpoint (though you might want the opposite permission flow - have folks who are pulling from the endpoint need to have permission to access the data your endpoint is making available - so we might consider giving Spade a UCAN that delegates ability to query our endpoint). And then when Spade confirms that they're taking a set of CommPs off for dealmaking, they can send a receipt letting us know to no longer include them in the stack (this can probably be a UCAN as well if we delegate them some `dealengine/pop` method or something like that)

Reply:
what they send us to confirm that the deal has been made is after that i guess. we'd at least need a receipt in UCAN form with the on-chain bits as proof (i don't think any actual "permissions" would necessarily need be invoked). but if we want it to be a permissioned flow, it could probably be similar to the above (we grant them permission to tell us a deal is made, they invoke this permission when it has, but they still should include the on-chain bits to make it so trust in spade isn't required to prove the deal was made)